app/models/ExportsDeclaration.scala

```diff
@@ -50,10 +50,9 @@ case class ExportsDeclaration(
 
   def isComplete: Boolean = status == DeclarationStatus.COMPLETE
 
-  def updatedItem(itemId: String, update: ExportItem => ExportItem): ExportsDeclaration =
-    itemBy(itemId).fold(this) { item =>
-      val updated = update(item)
-      copy(items = items.filterNot(_.id.equalsIgnoreCase(itemId)) + updated)
+  def updateItem(itemId: String, update: ExportItem => ExportItem): ExportsDeclaration =
+    itemBy(itemId).fold(this) { itemToReplace =>
+      copy(items = items.map(item => if (item.equals(itemToReplace)) update(itemToReplace) else item))
```

Comment:
item eq itemToReplace - nice chat for reference equality

Reply:
You're right.  Reference equality is good enough here.  I didn't even realise that scala had a third way of defining "equal" (i.e. 'eq') - I'll stick with '==' as an old Java hack.